src/main/java/com/bandwidth/sdk/messaging/models/MessageEvent.java

```diff
@@ -28,4 +41,28 @@ public boolean isMms(){
         return !isSms();
     }
 
+    @JsonIgnore
+    public boolean isError(){
+        return getErrorCode().isPresent();
+    }
+
+    @JsonIgnore
+    public Optional<MessageErrorTypes> getErrorType(){
+        Optional<MessageErrorTypes> errorType = Optional.empty();
+
+        if (getErrorCode().isPresent()){
```

Comment:
You can use `.map()` to simplify the logic here
https://docs.oracle.com/javase/8/docs/api/java/util/Optional.html#map-java.util.function.Function-